portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,58 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/**
+  * This is the main javascript file for the portfolio page. Contains a showProjectDetail method.
+  *
+  * Copyright 2019 Google LLC
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *  https://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  *
+  * @author: shradha-khapra
+  */
 
 /**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+  * Shows details of a specified project.
+  * @param {!Event} event The event object received.
+  * @return {void}
+  */
+function showProjectDetail(event) {
+  const descriptions =
+      [
+       "A Machine Learning Bot classifier used for segregation of Bots and Humans on Twitter during Indian socio-political uproar.",
+       "A Machine Learning image classifier to compare accuracy of KNN (with SVM) model and Neural Networks model for character recognition.", 
+       "A portal for organizing information and making appointments with healthcare professionals.", 
+       "A Google Maps & Weather API based application for real-time weather info on Maps with additional functionalities."
+      ];
+
+  // Add description to the page. 
+  const descriptionDiv = document.createElement('div');
+  const targetElement = event.target;
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  switch(targetElement.id) {
+    case "1":
```

Comment:
Now that you are using numbers instead of alphabets, you can simply use the indices for the map.

```
const index = parseInt(targetElement.id, /* radix= */ 10);
if(index < 0 || index >= descriptionDiv.length) {
  throw new Error("Description is not available for the given id.");
}
descriptionDiv.innerHTML = descriptions[index];
```

Also note that if your function throws an error, please annotate the function in the doc string.

```
/**
 * ...
 * ...
 * @throws {!Error} If the HTML id specified in the target element is incorrect.
 */
```